### ISSUE TITLE
support tensorflow 2.17+

### ISSUE THREAD

**Issue by ne1s0n** [autonomio/talos#602]
Currently `pip install talos` forces to downgrade TF to 2.14. This happens especially when using google colab, which by default has TF 2.17.
 
It's not technically a bug, yet, but it is pushing Talos toward senescence.  For example, I used to suggest using Talos in my courses on deep learning, but it's becoming harder to use it, since downgrading TF takes time, requires to restart the colab instance, and it gives a hacky feeling.

**Comment by mikkokotila** [comment]
Yes, this is not acceptable behaviour. We have to make it so that Talos does not take opinion on the TF version, at least not in terms of "greater than x version". Thanks for flagging this.
